sql/create.sql

```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
-	PRIMARY KEY (uname, rcode, sdatetime),
-	CHECK (amt > 0),								/* Check if amt is valid */
-	CHECK (sdatetime >= current_timestamp)			/* Check if the timestamp is valid */
+	expdate timestamp DEFAULT 						/* Expiry date */
+		(
+			current_timestamp + interval '3 months'
+		),
+	redeemed boolean DEFAULT FALSE,				/* States if the discount has been redeemed */
+	PRIMARY KEY (uname, rcode, expdate),
+	CHECK (expdate - interval '3 months' >= current_timestamp)			/* Check if the timestamp is valid */
 );
 
-CREATE TABLE Bids (
+CREATE TABLE Bid (
 	puname varchar(15),
 	duname varchar(15),
 	plate_no integer,
 	origin varchar(20),
 	dest varchar(20),
 	ptime time,
-	pdate date,
+	pdate date,	
 	won boolean DEFAULT FALSE,						/* States if the passenger has won the bid */
 	price integer NOT NULL,							/* Price of the bid */
+	btime time current_time,						/* Time of bid, updates when User updates Bid to change the bid price */
 	FOREIGN KEY (puname)
 		REFERENCES Passenger (uname)
 		ON DELETE CASCADE,
 	FOREIGN KEY (duname, plate_no, origin, dest, ptime, pdate)
 		REFERENCES Ride (uname, plate_no, origin, dest, ptime, pdate)
 		ON DELETE CASCADE,
 	PRIMARY KEY (puname, duname, plate_no, origin, dest, ptime, pdate),
+	CHECK (ptime > current_time + INTERVAL '1 hour'), /* Check that time passenger bids is 1hr before ride */
 	CHECK (puname <> duname),						/* Make sures that the passenger and driver are not the same person */
 	CHECK (price > 0)								/* Checks if the price is valid */
 );
-
 /*---------------------------- 		TRANSACTION PART		 ----------------------------------------*/
 
-CREATE TABLE Payment (
-	pcode char(7) PRIMARY KEY,						/* Payment code (Eg. ABC0000) */
-	pid SERIAL NOT NULL,							/* Candidate key: Payment ID */
-	CHECK (											/* CH = Cash, CD = Card */
-			pcode LIKE 'CH%'
-		   	OR pcode LIKE 'CD%'
-		  )
-);
-
-CREATE TABLE Cash (									/* Cash payment for the ride */
-	pcode char(7) PRIMARY KEY REFERENCES Payment
-		ON DELETE CASCADE
-);
 
-CREATE TABLE Card (									/* Card payment for the ride */
-	pcode char(7) PRIMARY KEY REFERENCES Payment
-		ON DELETE CASCADE,
-	surcharge numeric(2, 2) NOT NULL,						/* Extra xx.xx% of the total bill for administrative fees of using a card payment */
-	ctype varchar(15) NOT NULL,						/* Type of card payment */
-	company varchar(10) NOT NULL,					/* Card company */
-	CHECK (											/* Check if surcharge is valid: Maximum of 50% surcharge */
-			surcharge >= 0.00
-		   	OR surcharge <= 50.00
-		  ),
-	CHECK (											/* Check if ctype is valid: System only applicable for nets/credit/paywave transactions*/
-			ctype = 'NETS'
-		   	OR ctype = 'CREDIT'
-			OR ctype = 'PAYWAVE'
-		  ),
-	CHECK (											/* Check if company is valid: System only applicable for Master/Visa transactions */
-			company = 'MASTER'
-		   	OR company = 'VISA'
-		  )
-);
 
-CREATE TABLE Transacts (
+CREATE TABLE Transactions (
+	tcode varchar(15) PRIMARY KEY,
```

Comment:
why not reference a Ride instead?
Doesn't a ride and passenger unambiguously identify a transaction?